sugar-iconify.md

```diff
@@ -0,0 +1,229 @@
+see the tutorial on [Making Sugar
+Icons](http://wiki.laptop.org/go/Making_Sugar_icons).
+
+In some cases, you may desire to update a single icon, or a subset of
+the icons defined within a given SVG. To do so, pass a pattern to the
+script with -p, and only those icons which match the pattern will be
+exported.
+
+###Usage
+
+Download SVG by :
+
+```
+sugariconify.py [-ceghipv] [-s stroke_hex] [-f fill_hex] [-m | -o ] [-d directory] input.svg
+```
+Example usage: 
+
+```
+python sugariconify.py file.svg
+```
+
+Options
+```
+   -c    Apply default color entities (#666666, #FFFFFF) to output 
+   -d    [directory] The preferred output directory 
+   -e    Do not insert entities for strokes and fills 
+   -f    [hex] Hex value to replace with fill entity 
+   -g    Automatically accept guesses for stroke and fill entities 
+   -h    Display this help message 
+   -i    Insert "isolated stroke" entities 
+   -m    Multiple export; export top level groups as separate icons 
+   -o    Modify input file in place, overwriting it; overridden by -m 
+   -p    [pattern] Only export icons whose name contains pattern 
+   -s    [hex] Hex value to replace with stroke entity 
+   -v    Verbose 
+   -x    Create HTML preview files, for icon validation 
+```
+Note: A common problem encountered when using this script is when the
+target SVG has more than two colors.
+
+###Details
+ 
+Given a simple input SVG:
+ 
+```
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg
+   xmlns:dc="http://purl.org/dc/elements/1.1/"
+   xmlns:cc="http://creativecommons.org/ns#"
+   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+   xmlns:svg="http://www.w3.org/2000/svg"
+   xmlns="http://www.w3.org/2000/svg"
+   version="1.1"
+   viewBox="0 0 55 55"
+   height="55mm"
+   width="55mm">
+  <rect
+      y="2.5"
+      x="2.5"
+      height="50"
+      width="50"
+      style="fill:#ffff00;fill-opacity:1;stroke:#ff0000;stroke-width:3.5;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1" />
+</svg>
+```
+ 
+The resulting output would look like this:
+ 
+```
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!DOCTYPE svg  PUBLIC '-//W3C//DTD SVG 1.1//EN'
+          'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd' [
+        <!ENTITY stroke_color "#666666">
+        <!ENTITY fill_color "#ffffff">
+]>
+<svg
+   xmlns:dc="http://purl.org/dc/elements/1.1/"
+   xmlns:cc="http://creativecommons.org/ns#"
+   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+   xmlns:svg="http://www.w3.org/2000/svg"
+   xmlns="http://www.w3.org/2000/svg"
+   version="1.1"
+   viewBox="0 0 55 55"
+   height="55mm"
+   width="55mm">
+     <rect
+  <rect
+      y="2.5"
+      x="2.5"
+      height="50"
+      width="50"
+      style="fill:&fill_color;;fill-opacity:1;stroke:&stroke_color;;stroke-width:3.5;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"/>
+</svg>
+```
+##Instantiating SugarIconify
```

Comment:
I think this is a important section (Instantiating), because I separately didn't instantiate in each method.